src/service/CinemaService.java

```diff
@@ -0,0 +1,179 @@
+            String[] row = new String[cinema.getColumns()];
+            Arrays.fill(row, S);
+            seats[i] = row;
+        }
+
+        cinema.setSeats(seats);
+    }
+
+    public void showMenu() {
+        while (true) {
+            System.out.print("""
+                    1. Show the seats
+                    2. Buy a ticket
+                    3. Statistics
+                    0. Exit
+                    """);
+
+            String input = scanner.next();
+            System.out.println();
+
+            switch (input) {
+                case "0" -> {
+                    scanner.close();
+                    return;
+                }
+                case "1" -> showSeats();
+                case "2" -> buyTicket();
+                case "3" -> statistics();
+                default -> System.out.printf("Unknown menu item - %s\n", input);
+            }
+        }
+    }
+
+    private void showSeats() {
+        System.out.print("Cinema:\n ");
+        IntStream.rangeClosed(1, cinema.getSeats()[0].length)
+                .mapToObj(i -> " \t" + i)
+                .forEach(System.out::print);
+
+        System.out.println();
+
+        int counter = 1;
+        for (String[] seat : cinema.getSeats()) {
+            System.out.print(counter);
+            for (String s : seat) {
+                System.out.print("\t".concat(s));
+            }
+            System.out.println();
+            counter++;
+        }
+
+        System.out.println();
+    }
+
+    private void buyTicket() {
+        try {
+            Cinema.Seat seat = takeSeat();
+            seatValidation(seat);
+            int price = getPrice(seat);
+
+            cinema.setCurrentIncome(cinema.getCurrentIncome() + price);
+            cinema.setSoldTickets(cinema.getSoldTickets() + 1);
+
+            printPrice(price);
+        } catch (AlreadyTakenSeatException e) {
+            System.out.println("That ticket has already been purchased!\n");
+            buyTicket();
+        } catch (NonExistingSeatException e) {
+            System.out.println("Wrong input!\n");
+            buyTicket();
+        }
+    }
+
+    private void statistics() {
+        System.out.printf("""
+                Number of purchased tickets: %d
+                Percentage: %s%%
+                Current income: $%d
+                Total income: $%d
+                """, cinema.getSoldTickets(), getPercentage(), cinema.getCurrentIncome(), getTotalIncome());
+        System.out.println();
+    }
+
+    private Cinema.Seat takeSeat() {
+        System.out.println("Enter a row number:");
+        int row = scanner.nextInt();
+
+        System.out.println("Enter a seat number in that row:");
+        int column = scanner.nextInt();
+
+        System.out.println();
+        return new Cinema.Seat(row, column);
+    }
+
+    private void seatValidation(Cinema.Seat seat) throws NonExistingSeatException, AlreadyTakenSeatException {
+        final String B = "B";
```

Comment:
That's not a constant. Can you please use a more meaningful name than **B**?

Reply:
Fixed